src/components/Text/types.ts

```diff
@@ -0,0 +1,26 @@
+import React, { ReactNode } from "react";
+
+type HeadingTextTypes = 'H1' | 'H2' | 'H3' | 'H4' | 'H5' | 'H6';
+type BodyTextTypes = 'p+' | 'p' | 'p-';
+
+export type TextProps = {
+  /**
+   * Custom class(es) to apply
+   */
+  className?: string;
+
+  /**
+   * Custom styles to apply
+   */
+  style?: React.CSSProperties;
+
+  /**
+   * Main text content to render
+   */
+  text: ReactNode;
+
+  /**
+   * Predefined text type
+   */
+  type: HeadingTextTypes | BodyTextTypes;
+};
```

Comment:
can we put this file in a types folder?

Reply:
I find it easier to navigate supporting files for components when they are all located in the same place. E.g. im working on X component and I know i can find all the styles/types/util functions etc within the X directory. As we add more types i agree a higher level types folder would be useful for ones that are used more globally!

Reply:
fashooo - could we at least name this file something other than types.ts (perhaps ComponentName.ts) so we can avoid having lots of files with the same name?